### ISSUE TITLE
Darksiders III (606280)

### ISSUE THREAD

**Issue by mindinsomnia** [ValveSoftware/Proton#2357]
# Compatibility Report
- Name of the game with compatibility issues: Darksiders III
- Steam AppID of the game: 606280

## System Information
- GPU: GTX 980
- Driver/LLVM version: NVIDIA 415.27
- Kernel version: 4.15.0-45-generic
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/mindinsomnia/832896dbd6250bcd32acf08c71d3ad47
- Proton version: Beta (3.16-7)

## I confirm:
- [✓] that I haven't found an existing compatibility report for this game.
- [✓] that I have checked whether there are updates for my system available.

Game Log from a session that ended in a freeze/crash:
[steam-606280.log](https://github.com/ValveSoftware/Proton/files/2894159/steam-606280.log)

## Symptoms 
Basically, if you run the game for long enough you encounter a freeze that hangs for about 30 seconds before the game closes itself. These freezes and crashes occur about every hour of gameplay.

I personally haven't noticed any other issues so far in my 7 hours of gameplay.

## Reproduction
Play the game for any length of time.

**Comment by fsyy** [comment]
here the game crashes after the intro (doesn't matter if Proton 3.16 oder 4.2)

GPU: GTX 1060 6GB
Driver/LLVM version: 430.34
Kernel Version: 5.1.18-1-ck-nehalem
System Info: https://gist.github.com/fsyy/5b3a88721aa6c121d9fe5f6055e11f08
Proton Version: 4.2-9
Steam Log: https://gist.github.com/fsyy/adf4017ca25d6d5ce342bd32fdb0606a

**Comment by howdev** [comment]
This game works with great performance on Proton 4.11-6. Gold in ProtonDB

**Comment by kisak-valve** [comment]
Controller connect before game start cause game fail

Issue transferred from https://github.com/ValveSoftware/Proton/issues/3153.
@howdev posted on 2019-10-16T01:28:05:

# Compatibility Report
- Darksiders 3
- Steam AppID of the game: 606280

## System Information
- GPU: 1060
- Driver: Nvidia 340.40
- Kernel version: <4.15>
- Proton version: 4.11-7

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [ ] that I have checked whether there are updates for my system available.



## Symptoms 
dualshock 4 Bluetooth. When controller connected before starting the game, Steam exists and the game trying to continue but freezes, need to force close the game. 

The game starts OK without the controller connected, then connect the controller, the game continued to work.

## Reproduction
Using Steam Overlay and game controller option "forced on"
Start Steam
Turn on Dualshock 4 controller using Bluetooth connection
Start Darksiders 3
[steam-system.log](https://github.com/ValveSoftware/Proton/files/3732016/steam-system.log)
[steam-606280.log](https://github.com/ValveSoftware/Proton/files/3732017/steam-606280.log)

**Comment by kisak-valve** [comment]
Hello @howdev, just to confirm, was the game running fine with Proton 4.11-6 and a controller plugged in before starting the game?

**Comment by howdev** [comment]
It seems 4.11-6 may not have the issue. I cannot confirm since I cannot use 6 anymore? Steam will automatically update Proton

**Comment by alasky17** [comment]
@howdev -- I have a couple of points of clarification regarding your setup since I have not been able to reproduce this failure.

> When controller connected before starting the game, Steam exists and the game trying to continue but freezes, need to force close the game.

Does the game fail when the game window first launches or when you actually try to go in-game from the game menu?

> Using Steam Overlay 

Do you just mean that Steam Overlay is turned on in the options?  
> and game controller option "forced on"

Do you mean that you have "Playstation configuration support" turned on under the main Steam settings -> Controller -> General Controller settings? Or do you mean something else?
> Start Steam
> Turn on Dualshock 4 controller using Bluetooth connection

Are you using the official PS4 bluetooth dongle or a generic bluetooth dongle?

**Comment by howdev** [comment]
The game fails before is completely started, because Steam already exited. I am not sure what is happening here. But should be able to tell what is going on in the log. I have attached another log when the game is working. Should be able to compare the difference.

Steam Overlay is turned on in the Options.

"forced on" is the per game controller setting and not global. Cannot use global, since global will make all games detect the dualshock as xbox controller.

I have mentioned previously, the problem often happens but not always, that adds extra difficulty to finding the cause. 

Additional Information: The Tekken 7 works fine without using Steam controller support settings. 
So I suspect something to do with Proton controller with Darksiders 3 need to use Steam controller support for dualshock while Tekken 7 doesn't. 

There is no USB dongle. The computer supports Bluetooth. 
[steam-606280-working.log](https://github.com/ValveSoftware/Proton/files/3741745/steam-606280-working.log)

**Comment by alasky17** [comment]
@howdev Thank you for the clarification.  Since Steam itself is exiting, it leads me to another question - are you perchance using the Steam Beta client?  The Beta client currently has a known issue where several games cause the client itself to exit when they are launched.

**Comment by NTMan** [comment]
After passing the first boss, the game began to crash periodically.
1. [steam-606280.log](https://github.com/ValveSoftware/Proton/files/3774598/steam-606280.log)
2. [steam-606280.log](https://github.com/ValveSoftware/Proton/files/3774607/steam-606280.log)
3. [steam-606280.log](https://github.com/ValveSoftware/Proton/files/3774616/steam-606280.log)

etc ...


```
Unhandled exception: page fault on read access to 0x7f56cf659862 in 64-bit code (0x00007f56ad2e5299).
00dc:fixme:dbghelp:elf_search_auxv can't find symbol in module
Register dump:
 rip:00007f56ad2e5299 rsp:0000000018cba3e8 rbp:00007f56af7ca100 eflags:00010216 (  R- --  I   -A-P- )
 rax:0000000000000000 rbx:0000000000000024 rcx:0000000000000001 rdx:000000000000008d
 rsi:0000000000000128 rdi:00007f568c410b88  r8:00007f568c700678  r9:0000000000007f56 r10:000000000000feac
 r11:0000000000000008 r12:00007f56af797b00 r13:0000000000000000 r14:000000000ff47bb1 r15:00007f56cf659862
Stack dump:
0x0000000018cba3e8:  00007f568d028d00 0000000000000eb3
0x0000000018cba3f8:  00007f568c410bc8 00007f568c700640
0x0000000018cba408:  00007f568c6a1c90 00007f568c410b40
0x0000000018cba418:  00007f56aebae63e 0000000018cba5c0
0x0000000018cba428:  00007f568c410d10 00007f568c700640
0x0000000018cba438:  00007f568d0286a0 00007f56afdda7e0
0x0000000018cba448:  00007f568c9d5f80 00007f568cf2daa8
0x0000000018cba458:  0000000018cba500 0000000018cba5b0
0x0000000018cba468:  0000000018cba4e0 00007f568cf2d4b0
0x0000000018cba478:  00007f568c6192d8 000000010000068a
0x0000000018cba488:  00007f560000068a 00007f568d0288e8
0x0000000018cba498:  00007f568c6192f8 0000000200000002
Backtrace:
=>0 0x00007f56ad2e5299 _ZN4llvm11ms_demangle9Demangler17demangleClassTypeERNS_16itanium_demangle10StringViewE.cold+0x4e0909() in libllvm-10svn.so (0x00007f56af7ca100)
0x00007f56ad2e5299 _ZN4llvm11ms_demangle9Demangler17demangleClassTypeERNS_16itanium_demangle10StringViewE.cold+0x4e0909 in libllvm-10svn.so: addw	(%r15),%r10w
```

**Comment by howdev** [comment]
> @howdev Thank you for the clarification. Since Steam itself is exiting, it leads me to another question - are you perchance using the Steam Beta client? The Beta client currently has a known issue where several games cause the client itself to exit when they are launched.

No I am not running beta client

**Comment by howdev** [comment]
@NTMan you are using kernel 5.4?  llvm looks like AMDGPU driver.
seems like a driver problem. 
I have Nvidia and doesn't crash when the game is running. I already played 4 bosses.

**Comment by alasky17** [comment]
@howdev Thank you so much for confirming those things.  I tried again following your steps (5x times since you said it was inconsistent) but I am still not seeing Steam exiting before a window is created.  I have one last request in order to attempt to confirm if this is actually a regression from 4.11-6 to 4.11-7.    I didn't see you mention this earlier so forgive me if you have already tried -- could you try switching to Proton 4.2-9 and seeing if you are still able to trigger the crash?

**Comment by howdev** [comment]
Proton 4.2 the game starts without causing Steam to exit. but the controller does not always detected. 
So it seems that 4.11-6 should be working I remembered. So it may be a regression.

**Comment by alasky17** [comment]
@howdev Thank you for checking.  I'm at a bit of a loss because I cannot reproduce the crash, and there was nothing of note in the success and failure logs that you posted -- based on the logs it looks like it could be a Steam client bug rather than a Proton bug.  

Controller behavior changed a lot in Proton and is now much more correct than it was in 4.2.  For instance, needing Steam controller support on for Darksiders III with a ps4 dualshock controller is the correct behavior because the game does not support the ps4 dualshock on Windows and needs Steam controller support on there as well.  

There is still work being done to refine controller support, and hopefully the bug that you are running into will be fixed in the future, either by further wine/Proton work if that is the cause, or by fixing the bug in the Steam client if that is the cause of the crash.

**Comment by howdev** [comment]
@alasky17 There is new update on Steam client. Let's see what will happen.
I suppose we leave this for now until I have further information. 
Thank you for your efforts.

**Comment by howdev** [comment]
After the new Steam update, the is game is working with the controller connection without Steam crashes upon start

This game can be put on whitelist?

**Comment by alasky17** [comment]
@howdev Very glad to hear that!
